txcache/txListBySenderMap.go

```diff
@@ -7,43 +7,46 @@ import (
 	"github.com/multiversx/mx-chain-storage-go/txcache/maps"
 )
 
-const numberOfScoreChunks = uint32(100)
-
 // txListBySenderMap is a map-like structure for holding and accessing transactions by sender
 type txListBySenderMap struct {
-	backingMap        *maps.BucketSortedMap
+	backingMap        *maps.ConcurrentMap
 	senderConstraints senderConstraints
 	counter           atomic.Counter
-	scoreComputer     scoreComputer
 	txGasHandler      TxGasHandler
-	txFeeHelper       feeHelper
 	mutex             sync.Mutex
 }
 
 // newTxListBySenderMap creates a new instance of TxListBySenderMap
 func newTxListBySenderMap(
 	nChunksHint uint32,
 	senderConstraints senderConstraints,
-	scoreComputer scoreComputer,
 	txGasHandler TxGasHandler,
-	txFeeHelper feeHelper,
 ) *txListBySenderMap {
-	backingMap := maps.NewBucketSortedMap(nChunksHint, numberOfScoreChunks)
+	backingMap := maps.NewConcurrentMap(nChunksHint)
 
 	return &txListBySenderMap{
 		backingMap:        backingMap,
 		senderConstraints: senderConstraints,
-		scoreComputer:     scoreComputer,
 		txGasHandler:      txGasHandler,
-		txFeeHelper:       txFeeHelper,
 	}
 }
 
-// addTx adds a transaction in the map, in the corresponding list (selected by its sender)
-func (txMap *txListBySenderMap) addTx(tx *WrappedTransaction) (bool, [][]byte) {
+// addTxReturnEvicted adds a transaction in the map, in the corresponding list (selected by its sender).
+// This function returns a boolean indicating whether the transaction was added, and a slice of evicted transaction hashes (upon applying sender-level constraints).
+func (txMap *txListBySenderMap) addTxReturnEvicted(tx *WrappedTransaction) (bool, [][]byte) {
 	sender := string(tx.Tx.GetSndAddr())
 	listForSender := txMap.getOrAddListForSender(sender)
-	return listForSender.AddTx(tx, txMap.txGasHandler, txMap.txFeeHelper)
+	tx.precomputeFields(txMap.txGasHandler)
```

Comment:
This is the moment we set the computed fields of the "wrapped transaction". Maybe move this line to `txcache.AddTx`?

Reply:
Indeed, moved.

Also made the pre-computed fields atomic (maybe in some circumstances, the node might call `AddTx` concurrently for the same transaction - generally speaking, should not happen, though).